File: app/src/main/java/ru/sukharev/myrecipes/addrecipe/AddRecipeActivity.java
package ru.sukharev.myrecipes.addrecipe;

import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;

import ru.sukharev.myrecipes.R;


public class AddRecipeActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_recipe);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        AddRecipeFragment fragment = (AddRecipeFragment) getSupportFragmentManager().findFragmentById(R.id.add_recipe_fragment);
        AddRecipePresenter.init(this, fragment);
        FloatingActionButton floatingActionButton = findViewById(R.id.fab);
        floatingActionButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                try {
                    Listener listener = (Listener) AddRecipeActivity.this.getSupportFragmentManager().findFragmentById(R.id.add_recipe_fragment);
                    listener.onFabClick();
                } catch (ClassCastException ex) {
                    throw new AssertionError("inner fragment must implements activity listener");
                }
            }
        });
    }


    //used for passing activity events to fragments
    public interface Listener {

        void onFabClick();

    }

}

File: app/src/main/java/ru/sukharev/myrecipes/addrecipe/AddRecipeActivity.kt
package ru.sukharev.myrecipes.addrecipe

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_add_recipe.*
import ru.sukharev.myrecipes.R


class AddRecipeActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add_recipe)
        setSupportActionBar(toolbar)
        val fragment = supportFragmentManager.findFragmentById(R.id.add_recipe_fragment) as AddRecipeFragment
        AddRecipePresenter.init(this, fragment)
        fab.setOnClickListener {
            try {
                val listener = this@AddRecipeActivity.supportFragmentManager.findFragmentById(R.id.add_recipe_fragment) as Listener
                listener.onFabClick()
            } catch (ex: ClassCastException) {
                throw AssertionError("inner fragment must implements activity listener")
            }
        }
    }

    //used for passing activity events to fragments
    interface Listener {

        fun onFabClick()

    }

}
